components/user_groups_modal/ad_ldap_upsell_banner.tsx

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React, {useEffect, useState} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+import moment from 'moment';
+import {useIntl} from 'react-intl';
+
+import StartTrialBtn from 'components/learn_more_trial_modal/start_trial_btn';
+
+import {isAdmin} from 'mattermost-redux/utils/user_utils';
+import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
+import {getPrevTrialLicense} from 'mattermost-redux/actions/admin';
+import {getLicense} from 'mattermost-redux/selectors/entities/general';
+import {isCurrentLicenseCloud} from 'mattermost-redux/selectors/entities/cloud';
+import {getBrowserTimezone} from 'utils/timezone';
+
+import {LicenseLinks, LicenseSkus} from 'utils/constants';
+
+import {GlobalState} from '@mattermost/types/store';
+
+function ADLDAPUpsellBanner() {
+    const [show, setShow] = useState(true);
+    const [confirmed, setConfirmed] = useState(false);
+
+    const dispatch = useDispatch();
+    const {formatMessage} = useIntl();
+
+    useEffect(() => {
+        dispatch(getPrevTrialLicense());
+    }, []);
+
+    const isAdminUser = isAdmin(useSelector(getCurrentUser).roles);
+    const isCloud = useSelector(isCurrentLicenseCloud);
+    const currentLicense = useSelector(getLicense);
+    const prevTrialLicense = useSelector((state: GlobalState) => state.entities.admin.prevTrialLicense);
+    const prevTrialed = prevTrialLicense?.IsLicensed === 'true';
+
+    const isProfessional = currentLicense?.SkuShortName === LicenseSkus.Professional;
+
+    if (!show) {
+        return null;
+    }
+
+    const currentLicenseEndDate = new Date(parseInt(currentLicense?.ExpiresAt, 10));
+
+    const openSelfHostedLink = () => {
+        window.open(LicenseLinks.CONTACT_SALES, '_blank');
+    };
+
+    const confirmBanner = (
+        <div className='ad_ldap_upsell_confirm'>
+            <div className='upsell-confirm-backdrop'/>
+            <div className='upsell-confirm-foreground'>
+                <p className='title'>{formatMessage({id: 'adldap_upsell_banner.confirm.title', defaultMessage: 'Your trial has started!'})}</p>
+                <p className='subtitle'>{formatMessage({id: 'adldap_upsell_banner.confirm.license_trial', defaultMessage: 'Welcome to your Mattermost Enterprise trial! It expires on {endDate}. You now have access to high-security Enterprise features, for free.'}, {endDate: moment(currentLicenseEndDate).tz(getBrowserTimezone()).format('MMMM Do YYYY')})}</p>
```

Comment:
❓ Why not use `intl.formatDate` for the date format instead hardcoded a structure?